src/indexer/utils/evm/index.ts

```diff
@@ -130,10 +130,18 @@ export async function parseDeposit(
   }
 }
 
-export async function parseDestination(hexData: BytesLike, domain: Domain): Promise<string> {
+export async function parseDestination(hexData: BytesLike, domain: Domain, resourceType: string): Promise<string> {
   const arrayifyData = getBytes(hexData)
-  const recipientlen = Number("0x" + Buffer.from(arrayifyData.slice(32, 64)).toString("hex"))
-  const recipient = "0x" + Buffer.from(arrayifyData.slice(64, 64 + recipientlen)).toString("hex")
+  let recipient = ""
+  if (resourceType == ResourceTypes.FUNGIBLE || resourceType == ResourceTypes.NON_FUNGIBLE) {
```

Comment:
This is really a minor nitpick, but i just prefer switch here with a default that throws an error if resource type is not available.
Makes it a bit more readable and easier to add options in the future.